src/domain/repositories/RecipientRepository.ts

```diff
@@ -1,7 +1,8 @@
 import { Async } from "domain/entities/Async";
 import { Id } from "domain/entities/Base";
-import { Email } from "domain/entities/Notification";
+import { Email, UsernameEmail } from "domain/entities/Notification";
 
 export interface RecipientRepository {
     getByIds(ids: Id[]): Async<Email[]>;
+    getByUsernames(usernames: string[]): Async<UsernameEmail[]>;
```

Comment:
Typically, both `get` would return the same type, here we have `Email` and the new `UsernameEmail`. I think it's a good to have `UsernameEmail`, but with some refactors. I propose this general change:

- [ ] Rename RecipientRepository (and the implementation) to UserRepository. 
- [ ] Create entity User.ts (with id, username, and email).